public/app/plugins/datasource/prometheus/querycache/QueryCache.ts

```diff
@@ -62,26 +62,60 @@ function getTargSig(targExpr: string, request: DataQueryRequest<PromQuery>, targ
  * Ident: Identity: the string that is not expected to change
  * Sig: Signature: the string that is expected to change, upon which we wipe the cache fields
  */
-type Milliseconds = number;
 export class QueryCache {
-  private overlapWindow: Milliseconds;
+  private overlapWindowMs;
+  private perfObeserver: PerformanceObserver;
+  private pendingRequestIds = new Set<string>();
+
+  cache = new Map<TargetIdent, TargetCache>();
+
   constructor(overlapString?: string) {
     const unverifiedOverlap = overlapString ?? defaultPrometheusQueryOverlapWindow;
+
     if (isValidDuration(unverifiedOverlap)) {
       const duration = parseDuration(unverifiedOverlap);
-      this.overlapWindow = durationToMilliseconds(duration);
+      this.overlapWindowMs = durationToMilliseconds(duration);
     } else {
       const duration = parseDuration(defaultPrometheusQueryOverlapWindow);
-      this.overlapWindow = durationToMilliseconds(duration);
+      this.overlapWindowMs = durationToMilliseconds(duration);
     }
-  }
 
-  cache = new Map<TargetIdent, TargetCache>();
+    this.perfObeserver = new PerformanceObserver((list) => {
```

Comment:
IIUC... this intercept all resources after prometheus is initialized 🤔 it does seem like something we may want to move further up the stack for more general reporting (not just prometheus)

Reply:
yes, but i wanted to keep it isolated here for now. it't not clear to me if we're gonna get the green light to append requestId to the endpoint url. if that's a blocker, there's no need to spend time to make this generic.

Reply:
Yeah we'll probably wind up feature flagging this, and maybe that can get us the data we want without as much risk?